src/App.tsx

```diff
@@ -17,18 +49,43 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                todosStatus={todosStatus}
+                setTodosStatus={setTodosStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0
+                ? (
+                  <TodoList
+                    todos={filterTodo}
+                    selectedTodo={selectedTodo}
+                    setSelectedTodo={setSelectedTodo}
+                  />
+                )
+                : <Loader />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          setSelectedTodo={setSelectedTodo}
+          setHasLoadingError={setHasLoadingError}
+        />
+      )}
+
+      {hasLoadingError
+        && (
+          <div className="notification is-danger is-light">
+            Loading error
+          </div>
+        )}
```

Comment:
Loading error?))))))))
```suggestion
      {hasLoadingError && (
          <div className="notification is-danger is-light">
            Loading error
          </div>
        )}
```